### DISCUSSION TITLE
[FEATURE REQUEST] Support for deleting unconfigured lines in `file.keyvalue`

### DISCUSSION THREAD

**Discussion by tacerus** [saltstack/salt#67686] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

Currently the `file.keyvalue` state function only allows for either the modification of existing lines, or the addition of new lines.
It would be great if it could be used to prune lines not passed to be configured.

**Describe the solution you'd like**

Add an optional argument, such as `clean`, which, if set to `true`, enables the deletion of lines not passed via `key_values: {}`.

**Describe alternatives you've considered**

- Use additional states with dependency logic to prune the file before executing `file.keyvalue`.

- Use `file.managed` and construct the key value pairs manually .

**Additional context**

I will submit a proposed patch.

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by OrangeDog** [comment]
This is what `file.serialize` does when using the `keyvalue` serializer without the `merge_if_exists` option.
